MidasFrontend/feArduino.cxx

```diff
@@ -337,64 +342,44 @@ int Nloop, Ncount;
   return SUCCESS;
 }
 
-int dummy_counter = 0;
-
 /*-- Event readout -------------------------------------------------*/
 INT read_arduino_state(char *pevent, INT off)
 {
-
   // Create event header
   bk_init32(pevent);
-  
-  // Create a bank with some unsigned integers (move control variables))   
-  
-  // Bank data of unsigned int
-  uint32_t *pddata;
-  
-  // Bank names must be exactly four char
-  bk_create(pevent, "MOTO", TID_DWORD, (void**)&pddata);
-  
-  // Read the state of the motors from Arduino
-  // TOFIX!!!
-  uint32_t moving = 1;
-  uint32_t homing = 0;
-  uint32_t is_initialized = 0;
-  uint32_t at_limit = 0x0202;
-  uint32_t counter = dummy_counter++;
-  uint32_t gantry_position_x = (dummy_counter %100);  // Maybe these should be floats?
-  uint32_t gantry_position_y = 200 - (dummy_counter %100);  // Maybe these should be floats?
-
-  // Save variables in bank
-  *pddata++ = moving;
-  *pddata++ = homing;
-  *pddata++ = is_initialized;
-  *pddata++ = at_limit;
-  *pddata++ = counter;
-  *pddata++ = gantry_position_x;
-  *pddata++ = gantry_position_y;
-    
-  bk_close(pevent, pddata);	
 
-  // Create a bank with some float (temperature))  
-  
-  // Bank data of float
-  float *pddata2;
-  
-  // Bank names must be exactly four char
-  bk_create(pevent, "TEMP", TID_FLOAT, (void**)&pddata2);
-  
-  // Read the temperature from Arduino
-  // TOFIX!!!
-  float temp[5] = {34.2, 25.1, 34.4, 26.5, 34.3};
-  temp[3] += dummy_counter %4;
-  temp[4] -= dummy_counter %3;
-
-  // Save temperature variables in bank
-  for(int i = 0; i < 5; i++) *pddata2++ = temp[i];
-    
-  bk_close(pevent, pddata2);	
-  
-  
+  // Status Bank
+  DWORD status;
+  if (arduino_get_status(&status)) {
+    DWORD *pddata_status;
+    bk_create(pevent, ODB_BANK_STATUS, TID_DWORD, (void**)&pddata_status);
+    *pddata_status++ = status;
+    bk_close(pevent, pddata_status);
+  }
+
+  // Gantry Bank
+  float gantry_x_mm, gantry_y_mm;
+  if (arduino_get_position(&gantry_x_mm, &gantry_y_mm)) {
+    float *pddata_gantry;
+    bk_create(pevent, ODB_BANK_GANTRY, TID_FLOAT, (void**)&pddata_gantry);
+    *pddata_gantry++ = gantry_x_mm;
+    *pddata_gantry++ = gantry_y_mm;
+    bk_close(pevent, pddata_gantry);
+  }
+
+  // Temp Bank
+  TempData temp_data;
+  if (arduino_get_temp(&temp_data)) {
+    double *pddata_temp;
+    bk_create(pevent, ODB_BANK_TEMP, TID_DOUBLE, (void**)&pddata_temp);
+    *pddata_temp++ = temp_data.temp_ambient;
+    *pddata_temp++ = temp_data.temp_motor_x;
+    *pddata_temp++ = temp_data.temp_motor_y;
+    *pddata_temp++ = temp_data.temp_mpmt;
+    *pddata_temp++ = temp_data.temp_optical;
+    bk_close(pevent, pddata_temp);
+  }
+
```

Comment:
Do we store the actual user-input position in case there is a slight mismatch between distance traveled vs. desired? Wondering if that would go in the same bank as well?